packages/demo-app/src/App.tsx

```diff
@@ -1,66 +1,80 @@
-import { LogtoProvider, useLogto } from '@logto/react';
+import { LogtoProvider, useLogto, UserInfoResponse } from '@logto/react';
 import { signInNotificationStorageKey } from '@logto/schemas';
 import { demoAppApplicationId } from '@logto/schemas/lib/seeds';
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 
 import '@/scss/normalized.scss';
 import * as styles from './App.module.scss';
 import Callback from './Callback';
 import congrats from './assets/congrats.svg';
+import initI18n from './i18n/init';
+
+void initI18n();
 
 const Main = () => {
-  const { isAuthenticated, signIn } = useLogto();
+  const { isAuthenticated, fetchUserInfo, signIn, signOut } = useLogto();
+  const [user, setUser] = useState<UserInfoResponse>();
+  const { t } = useTranslation(undefined, { keyPrefix: 'demo_app' });
   const isInCallback = Boolean(new URL(window.location.href).searchParams.get('code'));
 
-  // Pending SDK fix
-  const username = 'foo';
-  const userId = 'bar';
-
   useEffect(() => {
     if (!isAuthenticated && !isInCallback) {
-      sessionStorage.setItem(
-        signInNotificationStorageKey,
-        'Use the admin username and password to sign in this demo.'
-      );
+      sessionStorage.setItem(signInNotificationStorageKey, t('notification'));
       void signIn(window.location.href);
     }
-  }, [isAuthenticated, isInCallback, signIn]);
+  }, [isAuthenticated, isInCallback, signIn, t]);
+
+  useEffect(() => {
+    (async () => {
+      if (isAuthenticated) {
+        const userInfo = await fetchUserInfo();
+        setUser(userInfo);
+      }
+    })();
+  }, [isAuthenticated, fetchUserInfo]);
 
   if (isInCallback) {
     return <Callback />;
   }
 
-  if (!isAuthenticated) {
+  if (!isAuthenticated || !user) {
     return null;
   }
```

Comment:
what would it look like if the user logged out?

Reply:
It will navigate you back to the sign in page